dio/lib/src/adapters/browser_adapter.dart

```diff
@@ -117,38 +117,40 @@ class BrowserHttpClientAdapter implements HttpClientAdapter {
     }
 
     final uploadStopwatch = Stopwatch();
-    xhr.upload.onProgress.listen((event) {
-      // This event will only be triggered if a request body exists.
-      if (connectTimeoutTimer != null) {
-        connectTimeoutTimer!.cancel();
-        connectTimeoutTimer = null;
-      }
-
-      final sendTimeout = options.sendTimeout;
-      if (sendTimeout != null) {
-        if (!uploadStopwatch.isRunning) {
-          uploadStopwatch.start();
+    if (requestStream != null) {
+      xhr.upload.onProgress.listen((event) {
+        // This event will only be triggered if a request body exists.
```

Comment:
@AlexV525 Regarding the timeouts, I think this comment is the key here. If there is no body, the following code never runs. But we should add a general test for that.